crates/ruff_linter/src/rules/ruff/rules/unnecessary_key_check.rs

```diff
@@ -0,0 +1,110 @@
+use ruff_python_ast::comparable::ComparableExpr;
+use ruff_python_ast::{self as ast, BoolOp, CmpOp, Expr};
+
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+
+/// ## What it does
+/// Checks for unnecessary key check before subscripting a dictionary.
+///
+/// ## Why is this bad?
+/// `get` can be used to get a value from a dictionary without having to check
+/// if the key exists first.
+///
+/// ## Examples
+/// ```python
+/// if "key" in dct and dct["key"]:
+///     ...
+/// ```
+///
+/// Use instead:
+/// ```python
+/// if dct.get("key"):
+///     ...
+/// ```
+#[violation]
+pub struct UnnecessaryKeyCheck {
+    key: String,
+    dict: String,
+}
+
+impl AlwaysFixableViolation for UnnecessaryKeyCheck {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Unnecessary key check")
+    }
+
+    fn fix_title(&self) -> String {
+        let UnnecessaryKeyCheck { key, dict } = self;
+        format!("Replace `{key} in {dict}` with `{dict}.get({key})`")
```

Comment:
nit: the highlighted range includes the entire boolean test

```suggestion
        format!("Replace with `{dict}.get({key})`")
```